src/mscorlib/shared/System/Text/StringBuilder.cs

```diff
@@ -2147,6 +2121,26 @@ unsafe private static void ThreadSafeCopy(char* sourcePtr, char[] destination, i
                 }
             }
         }
+
+        private static unsafe void ThreadSafeCopy(char[] source, int sourceIndex, char* destinationPtr, int destinationIndex, int count)
```

Comment:
It is weird that this overloads takes destinationIndex. It should just take destinationPtr and the caller should add destinationIndex to destinationPtr - to be consistent with the other overload that takes the source as pointer.